frontend/src/shared/helpers.tsx

```diff
@@ -1038,72 +1077,99 @@ export function prepareSampleMetadataForAgGrid(samples: Sample[]) {
 
 export function prepareSampleCohortDataForAgGrid(samples: Sample[]) {
   return samples.map((s) => {
-    const cohorts = s.cohortsHasCohortSampleConnection?.edges;
-    const cohortDates = cohorts?.flatMap((c) => {
-      return c.node.hasCohortCompleteCohortCompletes.map((cc) => {
-        return cc.date;
-      });
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      primaryId: sampleMetadata.primaryId,
+      cmoSampleName: sampleMetadata.cmoSampleName,
+      hasStatusStatuses: sampleMetadata.hasStatusStatuses,
+      revisable: s.revisable,
+      ...tempoData,
+    };
+  });
+}
+
+export function prepareCombinedSampleDataForAgGrid(samples: Sample[]) {
+  return samples.map((s) => {
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      revisable: s.revisable,
+      ...sampleMetadata,
+      ...tempoData,
+    };
+  });
+}
+
+function extractTempoFromSample(s: Sample) {
+  const cohorts = s.cohortsHasCohortSampleConnection?.edges;
+  const cohortDates = cohorts?.flatMap((c) => {
+    return c.node.hasCohortCompleteCohortCompletes.map((cc) => {
+      return cc.date;
     });
-    const deliveryDate = cohortDates?.sort()[0]; // earliest cohort date
+  });
+  let initialPipelineRunDate = cohortDates?.sort()[0]; // earliest cohort date
+  initialPipelineRunDate = formatCohortRelatedDate(initialPipelineRunDate);
 
-    let embargoDate;
-    if (deliveryDate !== undefined) {
-      let embargoDateAsDate = new Date(deliveryDate);
-      embargoDateAsDate.setMonth(embargoDateAsDate.getMonth() + 18);
-      embargoDate = moment(embargoDateAsDate).format("YYYY-MM-DD");
-    }
+  let embargoDate;
+  if (initialPipelineRunDate !== undefined) {
+    let embargoDateAsDate = new Date(initialPipelineRunDate);
+    embargoDateAsDate.setMonth(embargoDateAsDate.getMonth() + 18);
+    embargoDate = moment(embargoDateAsDate).format("YYYY-MM-DD");
+  }
 
-    const tempo = s.hasTempoTempos?.[0];
-    const { billedBy, costCenter, custodianInformation, accessLevel } =
-      tempo ?? {};
+  const tempo = s.hasTempoTempos?.[0];
+  const { billedBy, costCenter, custodianInformation, accessLevel } =
+    tempo ?? {};
```

Comment:
wow. fancy.  are you sure that it's not just ```tempo || {}``` ?

Reply:
I think both could work here. Looks like `??` lets us return `{}` if `tempo` is `undefined` or `null`, whereas `||` returns `{}` if `tempo` is anything falsy (broader cases).

I think `const tempo = s.hasTempoTempos?.[0];` could only either return some defined value or `undefined`, so I think `??` should be ok. 

Or should we use `||` to be safe?

Reply:
?? is fine.  i just hadn't seen it before.